key/aziot-keyd/src/lib.rs

```diff
@@ -71,23 +76,30 @@ pub async fn main(
             keys.set_parameter(&name, &value)?;
         }
 
-        Api { keys }
+        Api {
+            keys,
+            principals: principal_to_map(principal),
+        }
     };
     let api = std::sync::Arc::new(futures_util::lock::Mutex::new(api));
 
+    config_common::watcher::start_watcher(config_path, config_directory_path, api.clone());
+
     let service = http::Service { api };
 
     Ok((connector, service))
 }
 
 pub struct Api {
     keys: keys::Keys,
+    principals: std::collections::BTreeMap<libc::uid_t, Vec<wildmatch::WildMatch>>,
 }
 
 impl Api {
     pub fn new(
         aziot_keys: std::collections::BTreeMap<String, String>,
         preloaded_keys: std::collections::BTreeMap<String, String>,
+        principal: Vec<Principal>,
```

Comment:
Uhh, this `Api::new` fn is redundant with `main()` creating the `Api` directly. I probably refactored something in 5e2070f7c3c8fe03c1231883723622014d34f743 and forgot to delete this.

Just delete it and remove the `pub` from line 93.

Reply:
Done